ssl/ssl_lib.c

```diff
@@ -2024,10 +2049,14 @@ long SSL_ctrl(SSL *s, int cmd, long larg, void *parg)
             return 0;
     case SSL_CTRL_SET_MIN_PROTO_VERSION:
         return ssl_set_version_bound(s->ctx->method->version, (int)larg,
-                                     &s->min_proto_version);
+                                     &s->min_proto_version)
+               && ssl_check_allowed_versions(s->min_proto_version,
+                                             s->max_proto_version);
```

Comment:
I prefer that nothing is modified in the error case, and so that you first call ssl_check_allowed_versions().